src/modules/producer.js

```diff
@@ -8,6 +8,17 @@ const ERRORS = {
   BUFFER_FULL: 'Buffer is full'
 };
 
+class ProducerError extends Error {
+  constructor(error) {
+    super(error.message);
```

Comment:
You can use `Destructuring assignment` here

Reply:
changed